cpp/unit/ability.cpp

```diff
@@ -1,88 +1,242 @@
 // Copyright 2014-2015 the openage authors. See copying.md for legal info.
 
-#include "ability.h"
-
 #include <memory>
 
 #include "../terrain/terrain_object.h"
+#include "../datamanager.h"
+#include "ability.h"
 #include "action.h"
+#include "command.h"
 #include "unit.h"
 
 namespace openage {
 
-MoveAbility::MoveAbility(Texture *t, TestSound *s)
+std::string to_string(const ability_type &at) {
```

Comment:
I really like the idea of just using the std functions and be happy+surprised that it works even for custom types. But ok, if they say it's undefined when adding declarations to `std`, so be it. Then we probably should add these functions to `openage::util::`.

Template specializers are allowed though, so the custom `std::hash<>` functions can still go into `std`.